app/models/bitflyer_connection.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+class BitflyerConnection < ApplicationRecord
+  SUPPORTED_CURRENCIES = ["BAT", "USD", "BTC", "ETH"].freeze
+
+  belongs_to :publisher
+
+  validates :recipient_id, uniqueness: true, allow_blank: true
+
+  attr_encrypted :access_token, :refresh_token, key: :encryption_key
+
+  after_save :update_default_currency, if: -> { saved_change_to_default_currency? }
+
+  validates :default_currency, inclusion: { in: SUPPORTED_CURRENCIES }, allow_nil: true
+
+  after_destroy :selected_wallet_provider
+
+  def prepare_state_token!
+    update(state_token: SecureRandom.hex(64).to_s)
+  end
+
+  def payable?
+    true
+  end
+
+  def japanese_account?
+    country&.upcase == 'JP'
```

Comment:
Nit: Make this a constant at the top of the file, e.g. 

```
  JAPAN = "JP"
```
And do `country&.upcase == JAPAN`

Reply:
Fixed